winch/codegen/src/stack.rs

```diff
@@ -225,9 +233,22 @@ impl Stack {
         }
     }
 
-    /// Insert a new value at the specified index.
-    pub fn insert(&mut self, at: usize, val: Val) {
-        self.inner.insert(at, val);
+    /// Extend the stack with the given elements.
+    pub fn extend(&mut self, values: impl Iterator<Item = Val>) {
+        self.inner.extend(values);
+    }
+
+    /// Inserts many values at the given index.
+    pub fn insert_many(&mut self, at: usize, values: impl Iterator<Item = Val>) {
```

Comment:
Maybe this should be `impl IntoIterator<Item = Val>`? All the calls to it use `.into_iter()` on that argument.

Reply:
Thanks for the suggestion, fixed in https://github.com/bytecodealliance/wasmtime/pull/7155/commits/f4d975e4935ef46c48658bd6386707cdc8e6186d